cmd/api/src/api/middleware/auth.go

```diff
@@ -101,12 +101,28 @@ func AuthMiddleware(authenticator api.Authenticator) mux.MiddlewareFunc {
 
 // PermissionsCheck is a middleware func generator that returns a http.Handler which closes around a list of
 // permissions that an actor must have in the request auth context to access the wrapped http.Handler.
-func PermissionsCheck(authorizer auth.Authorizer, permissions ...model.Permission) mux.MiddlewareFunc {
+func PermissionsCheckAll(authorizer auth.Authorizer, permissions ...model.Permission) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+			if bhCtx := ctx.FromRequest(request); !bhCtx.AuthCtx.Authenticated() {
+				api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusUnauthorized, "not authenticated", request), response)
+			} else if !authorizer.AllowsAllPermissions(bhCtx.AuthCtx, permissions) {
+				api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusForbidden, "not authorized", request), response)
+			} else {
+				next.ServeHTTP(response, request)
+			}
+		})
+	}
+}
+
+// PermissionsCheck is a middleware func generator that returns a http.Handler which closes around a list of
+// permissions that an actor must have at least one in the request auth context to access the wrapped http.Handler.
+func PermissionsCheckAtLeastOne(authorizer auth.Authorizer, permissions ...model.Permission) mux.MiddlewareFunc {
```

Comment:
Since we are duplicating logic here, I was wondering if it would make sense to have a private function that takes in a permissions fn and then pass that to a generic handler which takes that function and runs that vs the 2 separate pathways. And then these 2 become simple wrappers calling it with the permissions function

Reply:
I don't necessarily disagree but I'm going to leave the code as-is just to keep things simple for myself. We would be better served having API side refactors handled by someone more experienced than I am!